### ISSUE TITLE
Scripts without either import or export are dropped from compilation

### ISSUE THREAD

**Issue by ChadKillingsworth** [google/closure-compiler#2535]
The new ES6 module force rewriting is no longer adding `goog.provide` calls and dependency information - the script is thus dropped.

**Comment by ChadKillingsworth** [comment]
Since we [have tests for this case](https://github.com/google/closure-compiler/blob/master/test/com/google/javascript/jscomp/CommandLineRunnerTest.java#L1852), I'm trying to figure out why it still drops the file.

**Comment by blickly** [comment]
Is it the --language_in=ES6?  I found it a bit confusing that commonjs code that doesn't use any other ES6 features can also be depending on ES6 transpilation.

**Comment by ChadKillingsworth** [comment]
Yes. I figured out what's happening. Dependency pruning is occurring before the goog.provide statement is added. The tests fail if the dependency mode is changed to `STRICT`.

**Comment by blickly** [comment]
Ah, whoops. Sorry about that!

**Comment by ChadKillingsworth** [comment]
ES6 modules that have `import` statements but no `export` statements aren't getting `goog.provide` calls added either.

**Comment by ChadKillingsworth** [comment]
This line is the culprit: https://github.com/google/closure-compiler/blob/master/src/com/google/javascript/jscomp/Es6RewriteModules.java#L473

An ES6 module which imports but doesn't export still needs a `goog.provide` statement - otherwise ProcessClosurePrimitives emits a "missing provide" error.

It would be nice if we could somehow mark those provides as synthetic so that ProcessClosurePrimitives didn't actually emit the namespace initialization statement.

**Comment by ChadKillingsworth** [comment]
This project makes a great test case: https://github.com/azakus/closure-reorder-import

**Comment by blickly** [comment]
But can't the namespace initialization be optimized away if it's unused?  I'd generally prefer for the module transpilation to be simpler and have a simple follow-up optimization over the transpilation having to keep track of this type of stuff.

**Comment by ChadKillingsworth** [comment]
I'd prefer that too - but right now it's creating errors that the namespace isn't defined.

**Comment by blickly** [comment]
I was under the impression that removing that conditional on L473 so that the provide is always created would be one way to fix this bug. But maybe I should reproduce the issue first.

**Comment by blickly** [comment]
I'm hoping this is fixed by https://github.com/google/closure-compiler/commit/b2a084fbedc3a071a1fecb44c3494d92ea19b92d

**Comment by rconnamacher** [comment]
I ran into an issue today importing a script with no exports in the latest Closure Compiler. Is this the same issue?

I'm importing an ES6 module that adds a method to the Promise.prototype, and so doesn't actually export anything.

```js
import "./util/PromiseFinally.js";
```

And in PromiseFinally.js:

```js
Promise.prototype.finally = function(handler) {
    // ...
}
```

When compiling: 
```
js/main.js:1: ERROR - required "module$js$util$PromiseFinally" namespace never provided
import "./util/PromiseFinally.js";
```

But simply adding `export default Promise;` to the end of PromiseFinally.js causes the build to pass.

Using Closure Compiler jar (from NPM) 20170806.0.0.

**Comment by ChadKillingsworth** [comment]
Closed by #2641
